specifyweb/frontend/js_src/lib/components/QueryBuilder/index.tsx

```diff
@@ -84,7 +84,7 @@ function QueryById({
   const recordSet = useQueryRecordSet();
 
   return query === undefined || recordSet === undefined ? null : (
-    <QueryBuilderWrapper query={query} recordSet={recordSet} />
+    <QueryBuilderWrapper key={queryId} query={query} recordSet={recordSet} />
```

Comment:
solution looks good and thanks for fixing this.

ideally, our component should be able to handle any prop changes without crashing - if you have time, and it doesn't look to completed, it would be better to modify the component to not crash, rather than nuke the component on pro pchange

Reply:
It was @melton-jason's idea for the key prop!

>  "don't change query on the fly" is like an implicit contract QueryBuilderWrapper puts out - this implicit contract is not guaranteed by TypeScript or tests

Can be done via ESLint? Exhaustive deps?

Reply:
add a required prop in the QueryBuilderWrapper component called "yesIAddedTheKeyProp"? 🤣

Reply:
crash during runtime if not. can't let anyone get away